internal/characters/chevreuse/burst.go

```diff
@@ -0,0 +1,83 @@
+package chevreuse
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+)
+
+var (
+	burstFrames []int
+)
+
+func init() {
+	burstFrames = frames.InitAbilSlice(59) // Q -> N1/Dash/Walk
+	burstFrames[action.ActionSkill] = 60
+	burstFrames[action.ActionJump] = 60
+	burstFrames[action.ActionSwap] = 59
+}
+
+const (
+	snapshotDelay         = 10
+	damageDelay           = 10
+	mineExplosionInterval = 60
+)
+
+func (c *char) Burst(p map[string]int) (action.Info, error) {
+
+	ai := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Explosive Grenade",
+		AttackTag:  attacks.AttackTagElementalBurst,
+		ICDTag:     attacks.ICDTagNone,
+		ICDGroup:   attacks.ICDGroupDefault,
+		StrikeType: attacks.StrikeTypeBlunt,
+		Element:    attributes.Pyro,
+		PoiseDMG:   100,
+		Durability: 50,
+		Mult:       burst[c.TalentLvlBurst()],
+	}
+
+	mineAi := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Secondary Explosive Shell",
+		AttackTag:  attacks.AttackTagElementalBurst,
+		ICDTag:     attacks.ICDTagChevreuseBurstMines,
```

Comment:
Should've asked earlier, what's the difference of ICDTag and ICDGroup?

Reply:
Two attacks can share the same ICDGroup (=use the same form of icd array/timer reset value) but have a different tag so that they are independent from each other, wikia's Internal Cooldown page explains it pretty well. There is no strict guideline on creating a new tag, but if it's possible to use a similar looking tag as long as it's not used in the char itself in another place which is what i suggested here